ydb/core/local_pgwire/local_pgwire_auth_actor.cpp

```diff
@@ -0,0 +1,186 @@
+#include "log_impl.h"
+#include "local_pgwire.h"
+#include "local_pgwire_util.h"
+
+#include <ydb/core/base/path.h>
+#include <ydb/core/base/ticket_parser.h>
+#include <ydb/core/grpc_services/local_rpc/local_rpc.h>
+#include <ydb/core/tx/scheme_cache/scheme_cache.h>
+
+#include <ydb/library/actors/core/actor.h>
+#include <ydb/library/actors/core/actor_bootstrapped.h>
+
+#include <ydb/public/api/grpc/ydb_auth_v1.grpc.pb.h>
+
+#include <ydb/services/persqueue_v1/actors/persqueue_utils.h>
+
+namespace NLocalPgWire {
+
+using namespace NActors;
+using namespace NKikimr;
+
+class TPgYdbAuthActor : public NActors::TActorBootstrapped<TPgYdbAuthActor> {
+    using TBase = TActor<TPgYdbAuthActor>;
+
+    struct TEvPrivate {
+        enum EEv {
+            EvTokenReady = EventSpaceBegin(NActors::TEvents::ES_PRIVATE),
+            EvAuthFailed,
+            EvEnd
+        };
+
+        static_assert(EvEnd < EventSpaceEnd(NActors::TEvents::ES_PRIVATE), "expect EvEnd < EventSpaceEnd(NActors::TEvents::ES_PRIVATE)");
+
+        struct TEvTokenReady : TEventLocal<TEvTokenReady, EvTokenReady> {
+            Ydb::Auth::LoginResult LoginResult;
+
+            TEvTokenReady() = default;
+        };
+
+        struct TEvAuthFailed : NActors::TEventLocal<TEvAuthFailed, EvAuthFailed> {
+            TString ErrorMessage;
+        };
+    };
+
+    TPgWireAuthData PgWireAuthData;
+    TActorId PgYdbProxy; 
+
+    TString DatabaseId;
+    TString FolderId;
+    TString SerializedToken;
+    TString Ticket;
+
+public:
+    TPgYdbAuthActor(const TPgWireAuthData& pgWireAuthData, const TActorId& pgYdbProxy)
+        : PgWireAuthData(pgWireAuthData)
+        , PgYdbProxy(pgYdbProxy) {
+    }
+
+    void Bootstrap() {
+        if (PgWireAuthData.UserName == "__ydb_apikey") {
+            SendDescribeRequest();
```

Comment:
may be move functions `SendDescribeRequest` and `SendApiKeyRequest` to some function "api key handler"?